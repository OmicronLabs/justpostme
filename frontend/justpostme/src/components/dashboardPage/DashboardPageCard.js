//@flow

import React from "react";
import styled from "styled-components";
import { IconButton } from "../common/Buttons";
import {
  PageBox,
  PageImage,
  PageTextContainer,
  PageInfoContainer,
  PageName,
  PageInfoItem,
  BlankPageBox,
  CreatePageIcon
} from "./DashboardPageCard.style";

export type CardProps = {
  card: Card
};

export type Card = {
  pageName: string,
  backgroundImage: string,
  scheduled: number,
  pending: number
};

export const GeneratedCard = (props: CardProps) => (
  <PageBox>
    <PageImage className="image" src={props.card.backgroundImage} />
    <PageTextContainer>
      <PageName className="name">{props.card.pageName}</PageName>
    </PageTextContainer>
    <PageInfoContainer>
      <PageInfoItem>pending: x</PageInfoItem>
      <PageInfoItem>scheduled: x</PageInfoItem>
    </PageInfoContainer>
  </PageBox>
);

export const AddPageCard = () => (
  <BlankPageBox>
    <IconButton>
      <CreatePageIcon className="fa fa-plus" />
    </IconButton>
  </BlankPageBox>
);
