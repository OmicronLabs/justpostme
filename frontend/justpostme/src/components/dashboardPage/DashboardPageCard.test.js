import React from "react";
import renderer from "react-test-renderer";

import { GeneratedCard, AddPageCard } from "./DashboardPageCard";
import testImage from "../../media/test_image.jpg";

import "jest-styled-components";

const testProps = {
  pageName: "testPage",
  backgroundImage: "../../media/test_image.png"
};

it("generates correct card from props", () => {
  const card = renderer.create(<GeneratedCard card={testProps} />).toJSON();
  expect(card).toMatchSnapshot();
});

it("generates correct /addcard/ card", () => {
  const addCard = renderer.create(<AddPageCard />).toJSON();
  expect(addCard).toMatchSnapshot();
});
