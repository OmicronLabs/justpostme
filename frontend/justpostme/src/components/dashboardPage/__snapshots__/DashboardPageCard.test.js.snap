// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates correct /addcard/ card 1`] = `
<div
  className="sc-dnqmqq iXhAvY"
>
  <a
    className="sc-htpNat hvFCZO"
  >
    <i
      className="fa fa-plus sc-VigVT kjYgMT"
    />
  </a>
</div>
`;
